File: ReleaseNotes.UI/src/services/project/models/release-notes/release-note-release-response-model.ts
import { ReleaseNotesReleaseNoteEntryResponseModel } from "./release-notes-release-note-entry-response-model";

export interface ReleaseNoteReleaseResponseModel {
  id: string;
  version: string;
  createdOnUtc: string;
  publishedOn: Date | undefined;
  noteEntries: ReleaseNotesReleaseNoteEntryResponseModel[];
}

File: ReleaseNotes.API/ReleaseNotes.API/Services/Project/Models/ReleaseNotes/ReleaseNoteReleaseResponseModel.cs
using ReleaseNotes.API.Enums;

namespace ReleaseNotes.API.Services.Project.Models.ReleaseNotes;

public class ReleaseNoteReleaseResponseModel
{
    public Guid Id { get; set; }
    public string Version { get; set; } = string.Empty;
    public DateTimeOffset CreatedOnUtc { get; set; }
    public DateTimeOffset? PublishedOn { get; set; }
    public IEnumerable<ReleaseNotesReleaseNoteEntryResponseModel> NoteEntries { get; set; } = [];
}
